pipelines/upgrade_pipeline.yml

```diff
@@ -0,0 +1,65 @@
+- hosts: localhost
+  vars_files:
+    - vars/{{ pipeline_type }}_base.yml
+    - vars/{{ pipeline_type }}_upgrade_base.yml
+  roles:
+    - forklift
+
+- hosts: pipeline-upgrade-{{ pipeline_type }}-{{ pipeline_version}}-{{ pipeline_os }}
+  become: yes
+  vars:
+  vars_files:
+    - vars/{{ pipeline_type }}_base.yml
+    - vars/{{ pipeline_type }}_base_release.yml
+    - vars/{{ pipeline_type }}_upgrade_{{ pipeline_version }}.yml
+    - vars/{{ pipeline_type }}_{{ pipeline_version_start }}.yml
+  roles:
+    - umask
+    - selinux
+    - etc_hosts
+    - foreman_server_repositories
+    - role: foreman_client_repositories
+      when: pipeline_version_start == "nightly" or pipeline_version_start is version('3.9', '>=')
+    - update_os_packages
+    - haveged
+    - disable_firewall
+    - foreman_installer
+
+- hosts: pipeline-upgrade-{{ pipeline_type }}-{{ pipeline_version}}-{{ pipeline_os }}
+  become: true
+  vars_files:
+    - vars/{{ pipeline_type }}_base.yml
+    - vars/{{ pipeline_type }}_base_release.yml
+    - vars/{{ pipeline_type }}_upgrade_{{ pipeline_version }}.yml
+    - vars/{{ pipeline_type }}_{{ pipeline_version_intermediate }}.yml
+  vars:
+    foreman_installer_upgrade: True
+    foreman_installer_skip_installer: "{{ true if pipeline_version_start == pipeline_version_intermediate else false }}"
```

Comment:
Not sure if Ansible can do this, but can it be `{{ pipeline_version_start == pipeline_version_intermediate }}`

Reply:
no idea, that's not my code here :eyes: